src/Icicle/Sea/FromAvalanche/Program.hs

```diff
@@ -205,7 +205,7 @@ seaOfXValue v t
       | ArrayT t' <- t
       -> let len = length vs
              writes arr (v',i)
-              = prim (PrimUpdate $ PrimUpdateArrayPut t')
+              = prim (PrimArray $ PrimArrayPutImmutable t')
```

Comment:
@amosr wait, on second thought.... are you sure about this? Buffers should be gone but arrays values can still occur here, I think?